File: ruby/countwords.rb
#!/usr/bin/env ruby

File.open("input.txt").read.split.inject({}) { |hash, word|
  hash[word] ||= 0
  hash[word] += 1
  hash
}.sort_by { |word, count|
  -count
}.each { |word, count|
  print "%s, %s\n" % [word, count]
}

File: python/countwords.py
#!/usr/bin/env python
# encoding: utf-8

from collections import Counter

def words(fileobj):
    for line in fileobj.readlines():
        for word in line.split():
            yield word

with open("input.txt", "r") as text:
    for entry in Counter(words(text)).most_common():
        print "%s, %d" % entry
